### ISSUE TITLE
after update bsl, restic backup is not work 

### ISSUE THREAD

**Issue by wgkingk** [vmware-tanzu/velero#5855]
**What steps did you take and what happened:**
1. backup with restic
2. update the bsl bucket to a new bucket 
3. backup again, The restic metadata was not backed up to the new bucket.

**What did you expect to happen:**

when change the bsl, and backup again, the restic metadata can be backup to the new bucket.

**Environment:**

- Velero version (use `velero version`): 
Client:
        Version: v1.10.0
        Git commit: 367f563072659f0bcd809bc33507fd75cd722344-dirty
Server:
        Version: v1.10.0
- Velero features (use `velero client config get features`): 
features: <NOT SET>
- Kubernetes version (use `kubectl version`):
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.8", GitCommit:"a12b886b1da059e0190c54d09c5eab5219dd7acf", GitTreeState:"clean", BuildDate:"2022-06-16T05:57:43Z", GoVersion:"go1.17.11", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.13", GitCommit:"592eca05be27f7d927d0b25cbb4241d75a9574bf", GitTreeState:"clean", BuildDate:"2022-10-12T10:50:48Z", GoVersion:"go1.17.13", Compiler:"gc", Platform:"linux/amd64"}

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

**Comment by Lyndon-Li** [comment]
From my understanding, this will not happen, because every time when a new podvolumebackup is created, Velero always refer to the current BSL to get the latest storage config.
However, there indeed some problems in this scenarios and we have discussed and fixed in [this issue](https://github.com/vmware-tanzu/velero/issues/5226).

**Comment by Lyndon-Li** [comment]
@wgkingk There may be some cases we have missed, if the problem indeed happen in your env, please provide us Velero log bundle.

**Comment by pdstefan** [comment]
I have a similar problem, though the scenario might differ. 
Scenario:  create 2 velero namespaces (velero-realtime and velero-schedules), each backed up by a different S3 provider. We are using minio in separate namespaces: 
- minio-realtime  with bucket name: velero
- minio-schedules with bucket name: backups 

Behavior:
We added velero-schedules first using helm (velero 0.10.1, node-agent enabled), created a BSL, used velero cli with no extra parameters to create a backup for a demo app using PVC in namespace `wikijs`. 
- Velero objects (kubernetes kinds mostly) were created in bucket `backups` in minio-schedules.minio-schedules.svc.cluster.local:9000. 
- restic repository  was initialized for `demo-app` 
- Backup maked "Completed"

Deployed velero-realtime using HELM (velero 0.10.1, node-agent enabled), created BSL pointing to minio-realtime.minio-realtime.svc.cluster.local:9000. Added extra parameter to velero cli to match the `velero-realtime` namespace ( `velero create backup -n velero-realtime  --include-namespaces wikijs real-3` ). 

- Velero objects were created in bucket `velero` in minio-realtime.minio-realtime.svc.cluster.local:9000
- Error when trying the FSB using restic: 
```
time="2023-02-09T15:58:12Z" level=error msg="Error checking repository for stale locks" backupRepo=velero-realtime/wikijs-default-restic-2x5sn error="error running command=restic unlock --repo=s3:http://minio-realtime.minio-realtime.svc.cluster.local:9000/velero/restic/wikijs --password-file=/tmp/credentials/velero-realtime/velero-repo-credentials-repository-password --cache-dir=/scratch/.cache/restic, stdout=, stderr=Fatal: unable to open config file: Stat: The specified key does not exist.\nIs there a repository at the following location?\ns3:http://minio-realtime.minio-realtime.svc.cluster.local:9000/velero/restic/wikijs\n: exit status 1" error.file="/go/src/github.com/vmware-tanzu/velero/pkg/repository/restic/repository.go:123" error.function="github.com/vmware-tanzu/velero/pkg/repository/restic.(*RepositoryService).exec" logSource="pkg/controller/backup_repository_controller.go:99"
time="2023-02-09T15:58:12Z" level=info msg="Running maintenance on backup repository" backupRepo=velero-realtime/wikijs-default-restic-2x5sn logSource="pkg/controller/backup_repository_controller.go:191"
time="2023-02-09T15:58:12Z" level=warning msg="error pruning repository" backupRepo=velero-realtime/wikijs-default-restic-2x5sn error="error running command=restic prune --repo=s3:http://minio-realtime.minio-realtime.svc.cluster.local:9000/velero/restic/wikijs --password-file=/tmp/credentials/velero-realtime/velero-repo-credentials-repository-password --cache-dir=/scratch/.cache/restic, stdout=, stderr=Fatal: unable to open config file: Stat: The specified key does not exist.\nIs there a repository at the following location?\ns3:http://minio-realtime.minio-realtime.svc.cluster.local:9000/velero/restic/wikijs\n: exit status 1" error.file="/go/src/github.com/vmware-tanzu/velero/pkg/repository/restic/repository.go:123" error.function="github.com/vmware-tanzu/velero/pkg/repository/restic.(*RepositoryService).exec" logSource="pkg/controller/backup_repository_controller.go:197"
time="2023-02-09T15:58:12Z" level=error msg="Error checking repository for stale locks" backupRepo=velero-realtime/wikijs-default-restic-2x5sn error="error running command=restic unlock --repo=s3:http://minio-realtime.minio-realtime.svc.cluster.local:9000/velero/restic/wikijs --password-file=/tmp/credentials/velero-realtime/velero-repo-credentials-repository-password --cache-dir=/scratch/.cache/restic, stdout=, stderr=Fatal: unable to open config file: Stat: The specified key does not exist.\nIs there a repository at the following location?\ns3:http://minio-realtime.minio-realtime.svc.cluster.local:9000/velero/restic/wikijs\n: exit status 1" error.file="/go/src/github.com/vmware-tanzu/velero/pkg/repository/restic/repository.go:123" error.function="github.com/vmware-tanzu/velero/pkg/repository/restic.(*RepositoryService).exec" logSource="pkg/controller/backup_repository_controller.go:99"
time="2023-02-09T15:58:12Z" level=info msg="Running maintenance on backup repository" backupRepo=velero-realtime/wikijs-default-restic-2x5sn logSource="pkg/controller/backup_repository_controller.go:191"
time="2023-02-09T15:58:12Z" level=warning msg="error pruning repository" backupRepo=velero-realtime/wikijs-default-restic-2x5sn error="error running command=restic prune --repo=s3:http://minio-realtime.minio-realtime.svc.cluster.local:9000/velero/restic/wikijs --password-file=/tmp/credentials/velero-realtime/velero-repo-credentials-repository-password --cache-dir=/scratch/.cache/restic, stdout=, stderr=Fatal: unable to open config file: Stat: The specified key does not exist.\nIs there a repository at the following location?\ns3:http://minio-realtime.minio-realtime.svc.cluster.local:9000/velero/restic/wikijs\n: exit status 1" error.file="/go/src/github.com/vmware-tanzu/velero/pkg/repository/restic/repository.go:123" error.function="github.com/vmware-tanzu/velero/pkg/repository/restic.(*RepositoryService).exec" logSource="pkg/controller/backup_repository_controller.go:197
``` 
- Backup marked as "PartiallyFailed". 

After spending some time debugging the problem I had a hunch that the restic component  in the `velero-realtime` namespace was using the incorrect credentials to connect to minio-realtime.minio-realtime.svc.cluster.local:9000. Tried different combinations with no success until I set the credentials for both s3 providers to be identical. 

Result was a bit confusing. When running `velero create backup -n velero-realtime  --include-namespaces wikijs real-3`:

- Velero objects were created in bucket `velero` in minio-realtime.minio-realtime.svc.cluster.local:9000
- Restic controller connected to minio-schedules.minio-schedules.svc.cluster.local:9000 and successfully initialized the restic repo. 
- Backup marked as "Complete" 

It is worth mentioning that the restic repo in minio-schedules.minio-schedules.svc.cluster.local:9000 was initially initialized by running `velero create backup --include-namespaces wikijs schedules-1`. 

To summarize: when credentials were the same for both s3 providers, restic backups triggered from "velero-realtime" were stored in the s3 bucket `backups` on `minio-schedules.minio-schedules.svc.cluster.local:9000`  instead of bucket `velero` on  `minio-realtime.minio-realtime.svc.cluster.local:9000`. 

Unfortunately, I dont know enough GO to figure out on my own where the problem is, but I am happy to help with any debugging or testing on my end (if helpful I can provide a bash script to setup the scenario described). 

Thank you.

**Comment by wgkingk** [comment]
> @wgkingk There may be some cases we have missed, or your current problem fell in one of the problems listed [here](https://github.com/vmware-tanzu/velero/issues/5226#issuecomment-1301601619). To confirm the cause, please provide us more info:
> 
> * The Velero log bundle
> * The status of the problematic backup, is it Complete or PartiallyFailed?

I noticed that this pr is not merged to 1.10，have a plan？

**Comment by Lyndon-Li** [comment]
@wgkingk Fix for #5226 will be included in v1.11
